lib/extensions/ar_virtual.rb

```diff
@@ -143,6 +143,7 @@ def define_delegate(method_name, method, to: nil, allow_nil: nil, default: nil)
       if allow_nil
         method_def = [
           "def #{method_name}(#{definition})",
+          "return self[:#{method_name}] if has_attribute?(:#{method_name})",
```

Comment:
Essentially, it allows you to get a valid stacktrace when something goes wrong, because it is defined on a single line.  I would say keep it as an array, because you don't want to be trudging through metaprogrammed defined line methods to figure out whats up, but up to you.